petastorm/spark/spark_dataset_converter.py

```diff
@@ -1,21 +1,39 @@
 from petastorm import make_batch_reader
+from petastorm.fs_utils import FilesystemResolver
+from petastorm.reader import normalize_data_url
 from petastorm.tf_utils import make_petastorm_dataset
 from pyspark.sql.session import SparkSession
 
 import atexit
 import os
+import pyarrow
 import shutil
 import threading
 import uuid
 
-DEFAULT_CACHE_DIR = "/tmp/spark-converter"
+DEFAULT_CACHE_DIR = "file:///tmp/spark-converter"
 ROW_GROUP_SIZE = 32 * 1024 * 1024
 
 
 def _get_spark_session():
     return SparkSession.builder.getOrCreate()
 
 
+def _delete_cache_data(dataset_url):
+    """
+    Get fs handler from java gateway
+    :return:
+    """
+    hdfs_driver = 'libhdfs3'
+    dataset_url = normalize_data_url(dataset_url, hdfs_driver)
+    resolver = FilesystemResolver(dataset_url, hdfs_driver)
+    fs = resolver.filesystem()
+    if isinstance(fs, pyarrow.LocalFileSystem):
+        shutil.rmtree(dataset_url[7:], ignore_errors=True)
```

Comment:
remove the "[7:]", rmtree should support path like `file:///...'

Reply:
oh... not support...

Reply:
Let's try whether it works:`fs.delete('file:///...')`

Reply:
If not work, use `shutil.rmtree.(urlparse(data_url).path)`
don't ignore error but let outer block catch error and print warning...

Reply:
fs.delete('file:///...') doesn't work. See the commit: delete doesn't work. I'll use shutil.